advisor/src/main/kotlin/Advisor.kt

```diff
@@ -67,11 +69,20 @@ class Advisor(
 
         val providers = providerFactories.map { it.create(config) }
 
+        val results = mutableMapOf<Identifier, MutableList<AdvisorResult>>()
+
         val packages = ortResult.getPackages(skipExcluded).map { it.pkg }
-        val results = runBlocking { providers.first().retrievePackageVulnerabilities(packages) }
-            .mapKeysTo(sortedMapOf()) { (pkg, _) -> pkg.id }
+        runBlocking {
```

Comment:
While this works, the code could be made more efficient by running the providers in parallel; as they are independent on each other, there should not be any conflicts between them. Instead of the `forEach` loop, use a `map` in combination with `async` to trigger the different providers asynchronously. Their results can then be merged after all have completed.

You can refer to `Analyzer.kt` for an example how this is done.

Reply:
Thanks for the idea! I also was not really satisfied with this block.
Have a look at my solution!